engine/time/src/main/java/io/deephaven/time/calendar/ReadOptimizedConcurrentCache.java

```diff
@@ -117,10 +117,10 @@ public V computeIfAbsent(int key) {
         final V newValue = valueComputer.apply(key);
 
         if (newValue == null) {
-            throw new IllegalArgumentException("Computed a null value: key=" + key);
+            throw new NullPointerException("Computed a null value: key=" + key);
         }
 
-        existing = cache.putIfAbsent(key, newValue);
+        existing = cache.put(key, newValue);
         return existing == null ? newValue : existing;
```

Comment:
If we're not doing `putIfAbsent` (which was my recommendation) we should prefer to return `newValue` here. In the event that a race does allow for more than one entry to have been created, the one in the map is the longer-lived, and we would prefer the other (`existing`) to be GC'd ASAP.